tools/install-source/Program.cs

```diff
@@ -25,87 +33,200 @@ static bool ParseBool (string value)
 			return bool.Parse (value);
 		}
 	}
-	public static void Main (string[] arguments)
+
+	public static string FixPathEnding(string path)
+	{
+		if (!path.EndsWith (Path.DirectorySeparatorChar.ToString (), StringComparison.Ordinal))
+			return path + Path.DirectorySeparatorChar;
+		return path;
+	}
+
+	// returns a tuple with the pdb files as the first value and the mdb files as the second
+	public static Tuple <HashSet<String>, HashSet<String>> GetSplittedPaths (List<string> paths)
+	{
+		var splittedPaths = new Tuple<HashSet<String>, HashSet<String>> (new HashSet<string> (), new HashSet<string> ());
+		foreach (var p in paths) {
+			if (p.EndsWith (".pdb", StringComparison.Ordinal)) {
+				splittedPaths.Item1.Add (p);
+				continue;
+			}
+			if (p.EndsWith (".mdb", StringComparison.Ordinal)) {
+				splittedPaths.Item2.Add (p);
+				continue;
+			}
+		}
+		return splittedPaths;
+	}
+
+	// returns the source paths used to create an assembly that has an mdb file
+	public static HashSet<String> GetFilePathsFromMdb (string mdb_file)
+	{
+		var srcs = new HashSet<String> ();
+		MonoSymbolFile symfile;
+
+		try {
+			symfile = MonoSymbolFile.ReadSymbolFile(mdb_file);
+			srcs.UnionWith (from src in symfile.Sources select src.FileName);
+
+		} catch (IOException ioe) {
+			Console.WriteLine("IO error while reading msb file '{0}': {1}", mdb_file, ioe.Message);
+		}
+
+		return srcs;
+	}
+
+	// returns the source paths used to create an assembly that has a pdb file
+	public static HashSet<String> GetFilePathsFromPdb (string pdbFile)
+	{
+		var assemblyPath = pdbFile.Replace (".pdb", ".dll");
+		var pdb = assemblyPath + ".pdb";
```

Comment:
This looks wrong, if you have `foo.dll`, the pdb is `foo.pdb`, not `foo.dll.pdb` (this is a change from mdb, where `foo.dll.mdb` existed).